quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -236,34 +230,51 @@ protected int runReAugmentationIfNeeded(List<String> cliArgs, CommandLine cmd, C
         return exitCode;
     }
 
+    protected void initProfile(List<String> cliArgs, String currentCommandName) {
+        if (currentCommandName.equals(StartDev.NAME)) {
+            // force the server image to be set with the dev profile
+            Environment.forceDevProfile();
+        } else {
+            String profile = org.keycloak.common.util.Environment.getProfile();
+            if (profile != null) {
+                Environment.setProfile(profile);
+            }
+
+            parseConfigArgs(cliArgs, (k, v) -> {
+                if (k.equals(Main.PROFILE_SHORT_NAME) || k.equals(Main.PROFILE_LONG_NAME)) {
+                    Environment.setProfile(v);        
+                }
+            }, ignored -> {});
+        }
+    }
+
     private static boolean shouldSkipRebuild(List<String> cliArgs, String currentCommandName) {
         return cliArgs.contains("--help")
                 || cliArgs.contains("-h")
                 || cliArgs.contains("--help-all")
                 || currentCommandName.equals(Build.NAME)
                 || currentCommandName.equals(ShowConfig.NAME)
-                || currentCommandName.equals(BootstrapAdmin.NAME)
-                || currentCommandName.equals(Tools.NAME);
+                || currentCommandName.equals(Completion.NAME);
```

Comment:
> So, as Tools command contains only the Completion subcommand, it will never be set as the current command - always the Completion is used. I understand it.

Yes, that is correct. Checking for the parent command here is ineffective.

> But what is the case for the bootstrap admin?

Similar - checking for the parent command here isn't doing anything.

> AFAIK, the NonServer commands never run augmentation, is it correct?

Unfortunately no. import, export, bootstrap-admin are all allowed to auto-build.